### PR TITLE
Implement New Weaviate Repository

### LINKED ISSUE #59: Update datastew version to use new weaviate repository

### PR DESCRIPTION
**Pull request by mehmetcanay**
Implemented the new weaviate repository functions for the API endpoints.

closes #59

### CODE REVIEW

File: api/routes.py [hunk 1]
```diff
@@ -123,17 +129,26 @@ async def get_all_concepts():
 
 
 @app.put("/concepts/{id}", tags=["concepts"])
-async def create_or_update_concept(id: str, terminology_id: str, name: str):
```
Comment (tiadams) [inline api/routes.py:126]: Unless the request param name matches the param name, this should not work afaik

Test for e.g.` /concepts/some_id `

File: api/routes.py [hunk 2]
```diff
@@ -143,33 +158,58 @@ async def get_all_mappings():
 
 
 @app.put("/concepts/{id}/mappings", tags=["concepts", "mappings"])
-async def create_concept_and_attach_mapping(id: str, terminology_id: str, concept_name: str, text: str):
+async def create_concept_and_attach_mapping(concept_id: str,  concept_name: str, terminology_name, text: str):
     try:
```
Comment (tiadams) [inline api/routes.py:162]: same here

File: api/routes.py [hunk 3]
```diff
@@ -143,33 +158,58 @@ async def get_all_mappings():
...
 
 @app.put("/mappings/", tags=["mappings"])
-async def create_or_update_mapping(concept_id: str, text: str):
+async def create_mapping(concept_id: str, text: str):
     try:
-        concept = repository.session.query(Concept).filter(Concept.id == concept_id).first()
-        if not concept:
+        if not repository._concept_exists(concept_id=concept_id):
             raise HTTPException(status_code=404, detail=f"Concept with id {concept_id} not found")
+        result = repository.client.query.get(
```
Comment (tiadams) [inline api/routes.py:192]: this logic should be part of datastew imo

File: api/routes.py (hunk 1 above)
Comment (mehmetcanay) [reply api/routes.py:126]: You are right, I missed it. Fixed in the next commit.

File: api/routes.py (hunk 2 above)
Comment (mehmetcanay) [reply api/routes.py:162]: And fixed here too.

File: api/routes.py (hunk 3 above)
Comment (mehmetcanay) [reply api/routes.py:192]: You're right, we can have functions like `get_terminology(terminology_name: str)` and `get_concept(concept_id: str)`. Should I start implementing it?

File: api/routes.py [hunk 4]
```diff
@@ -260,4 +262,4 @@ async def import_snomed_ct(background_tasks: BackgroundTasks):
 
 
 if __name__ == "__main__":
-    uvicorn.run(app, host="0.0.0.0", port=5000)
+    uvicorn.run(app, host="0.0.0.0", port=5001)
```
Comment (tiadams) [inline api/routes.py:265]: Please set this back to 5000

**Review by tiadams — CHANGES_REQUESTED** [review]

**Review by tiadams — APPROVED** [review]

**Pull request merged**